lib/dealership.rb

```diff
@@ -0,0 +1,53 @@
+class Dealership
+  attr_reader :name, :address, :inventory, :car
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    inventory.count
+  end
+
+  def add_car(car)
+    inventory << car
+  end
+
+  def has_inventory?
+    return false if inventory.count == 0
+  end
+
+  def cars_by_make(make)
+    inventory.find_all do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    inventory.sum do |car|
+      car.total_cost
+    end
+  end
+
+  def details
+    deatils = {
+      address: address,
+      total_value: total_value
+    }
+  end
+
+  def average_price_of_car
+    average = inventory.map do |car|
+      car.total_cost
+    end
+    average.sum / inventory.count
```

Comment:
interaction pattern was looking for a string to come out of this method, so you'd want to do a `.to_s` at the end of your line 45. You may also need to find a way to add a thousands' comma (39,000) where necessary. I4s will always push you a little bit. :)